### PR TITLE
Use setup-conda action

### PR DESCRIPTION
**Pull request by jonashaag**
- [ ] closes #xxxx (Replace xxxx with the Github issue number)
- [ ] [Tests added and passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#writing-tests) if fixing a bug or adding a new feature
- [ ] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).
- [ ] Added [type annotations](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#type-hints) to new arguments/methods/functions.
- [ ] Added an entry in the latest `doc/source/whatsnew/vX.X.X.rst` file if fixing a bug or adding a new feature.

### CODE REVIEW

**Comment by jonashaag** [comment]
@mroeschke

File: .github/actions/setup-conda/action.yml [hunk 1]
```diff
@@ -21,8 +27,10 @@ runs:
       uses: conda-incubator/setup-miniconda@v2.1.1
       with:
```
Comment (mroeschke) [inline .github/actions/setup-conda/action.yml:28]: Looks like this is missing the `use-only-tar-bz2: true` option

File: .github/workflows/asv-bot.yml [hunk 2]
```diff
@@ -41,13 +41,8 @@ jobs:
...
         # during discovery process
-      - uses: conda-incubator/setup-miniconda@v2.1.1
-        with:
-          activate-environment: pandas-dev
-          channel-priority: strict
-          environment-file: ${{ env.ENV_FILE }}
-          use-only-tar-bz2: true
-          condarc-file: ci/condarc.yml
+      - name: Set up Conda
+        uses: ./.github/actions/setup-conda
```
Comment (mroeschke) [inline .github/workflows/asv-bot.yml:45]: Nit: Could we rename this actions `setup_conda` to match `build_pandas`?

File: .github/workflows/sdist.yml [hunk 3]
```diff
@@ -59,12 +59,12 @@ jobs:
...
 
-    - uses: conda-incubator/setup-miniconda@v2.1.1
+    - name: Set up Conda
+      uses: ./.github/actions/setup-conda
       with:
-        activate-environment: pandas-sdist
-        channels: conda-forge
-        python-version: '${{ matrix.python-version }}'
-        condarc-file: ci/condarc.yml
+        environment-file: ""
```
Comment (mroeschke) [inline .github/workflows/sdist.yml:65]: Does this run correctly if empty string is provided?

File: .github/actions/setup-conda/action.yml (hunk 1 above)
Comment (jonashaag) [reply .github/actions/setup-conda/action.yml:28]: Thanks!

File: .github/workflows/asv-bot.yml (hunk 2 above)
Comment (jonashaag) [reply .github/workflows/asv-bot.yml:45]: The GHA convention seems to be `kebab-case`, see eg. https://github.com/conda-incubator/setup-miniconda/blob/main/action.yml#L23

I think we should stick to one style everywhere (action names, inputs, ...). Please make a decision if we want `snake_case` or `kebab-case`.

File: .github/workflows/sdist.yml (hunk 3 above)
Comment (jonashaag) [reply .github/workflows/sdist.yml:65]: Yes it is the default value, see https://github.com/conda-incubator/setup-miniconda/blob/e81abac10ce2c37423b54eae5af93aa3b4d3475c/action.yml#L78. I actually checked that `sdist` works in an earlier version of the PR by uncommenting the `if`.

File: .github/workflows/asv-bot.yml (hunk 2 above)
Comment (mroeschke) [reply .github/workflows/asv-bot.yml:45]: Okay happy to change `build_pandas` to `build-pandas` then (feel free to change this in one of your other PRs)

**Review by mroeschke — APPROVED** [review]
LGTM. Do you think it also makes sense to add the caching of conda packages to this step too?

```
    - name: Cache conda
      uses: actions/cache@v3
      env:
        CACHE_NUMBER: 0
      with:
        path: ~/conda_pkgs_dir
        key: ${{ runner.os }}-conda-${{ env.CACHE_NUMBER }}-${{
          hashFiles('${{ env.ENV_FILE }}') }}
```

**Comment by jonashaag** [comment]
Will send more PRs with caching changes soon :)

**Comment by mroeschke** [comment]
Thanks @jonashaag, looking forward to the followups :)

**Pull request merged**